OsmAnd-api/src/net/osmand/aidlapi/IOsmAndAidlInterface.aidl

```diff
@@ -860,4 +861,34 @@ interface IOsmAndAidlInterface {
      * @params keyEventList (List<Integer>) - list of requested key events
      */
     long registerForKeyEvents(in AKeyEventsParams params, IOsmAndAidlCallback callback);
-}
\ No newline at end of file
+
+    /**
+     * Method to get position of various objects
+     *
+     * @params positionType (int) - type of position to get
+     */
+    boolean getPosition(in int positionType, out APosition position);
+
+    /**
+     * Method to get index of the current route segment
+     */
+    int getCurrentRouteSegmentIndex();
+
+    /**
+     * Method to get creation time of current route
+     *
+     */
+    long getRouteCreationTime();
+
+    /**
+     * Method to get current route points
+     *
+     */
+    boolean getRoutePoints(out List<ALatLon> route);
+
+    /**
+     * Method to get application mode
+     *
+     */
+    String getApplicationMode();
+}
```

Comment:
It should be refactored into 3 differnet methods:
1. getCalculatedRoute - information about calculated routes, geometry, appmode, time, etc
2. getApplicationMode - current app mode
3. getNavigationStatus - progress, currentSegment, leftDistance, leftTime, etc

Reply:
Done in latest commit